lib/components/app.js

```diff
@@ -2,21 +2,69 @@ import React from 'react'
 import Peerlist from '../components/peerlist'
 import Tracklist from '../components/tracklist'
 import SearchBar from '../components/search-bar'
+import RestartNotifier from '../components/restart-notifier'
 import Player from '../components/player'
+import appStatusStore from '../stores/app-status-store'
 
 class App extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = appStatusStore.getState()
+  }
+
+  componentDidMount() {
+    appStatusStore.listen(state => this._onChange(state))
+  }
+
+  componentWillUnmount() {
+    appStatusStore.listen(state => this._onChange(state))
+  }
+
+  _onChange(state) {
+    this.setState(state)
+  }
+
   render () {
-    var wrapperStyle = { paddingBottom: 150 }
-    var style = { padding: '71px 0 0 0', margin: 'auto' }
+    var styles = {
+      container: {
+        display: 'flex',
+        flexDirection: 'column',
+        justifyContent: 'space-between',
+        height: '100%',
+      },
+      header: {
+        flex: '0 0 auto',
+      },
+      section: {
+        flex: '1 1 auto',
+        minHeight: 0,
+        overflow: 'auto',
+        display: 'flex',
+        alignContent: 'center',
+        padding: '0.5em',
+      },
+      tracklist: {
+        flex: '0 0 100%',
+      },
+      footer: {
+        flex: '0 0 auto',
+      },
+    }
```

Comment:
I like all the styles in a single object.  Big improvement over dealing with several objects or methods for styles.